openhands/resolver/send_pull_request.py

```diff
@@ -325,53 +309,46 @@ def send_pull_request(
 
     # Prepare the PR data: title and body
     final_pr_title = (
-        pr_title
-        if pr_title
-        else f'Fix issue #{github_issue.number}: {github_issue.title}'
+        pr_title if pr_title else f'Fix issue #{issue.number}: {issue.title}'
     )
-    pr_body = f'This pull request fixes #{github_issue.number}.'
+    pr_body = f'This pull request fixes #{issue.number}.'
     if additional_message:
         pr_body += f'\n\n{additional_message}'
     pr_body += '\n\nAutomatic fix generated by [OpenHands](https://github.com/All-Hands-AI/OpenHands/) 🙌'
 
     # If we are not sending a PR, we can finish early and return the
     # URL for the user to open a PR manually
     if pr_type == 'branch':
-        url = f'https://github.com/{push_owner}/{github_issue.repo}/compare/{branch_name}?expand=1'
+        url = handler.get_compare_url(branch_name)
     else:
         # Prepare the PR for the GitHub API
-        data = {
-            'title': final_pr_title,  # No need to escape title for GitHub API
-            'body': pr_body,
-            'head': branch_name,
-            'base': base_branch,
-            'draft': pr_type == 'draft',
-        }
-
-        # Send the PR and get its URL to tell the user
-        response = requests.post(f'{base_url}/pulls', headers=headers, json=data)
-        if response.status_code == 403:
-            raise RuntimeError(
-                'Failed to create pull request due to missing permissions. '
-                'Make sure that the provided token has push permissions for the repository.'
-            )
-        response.raise_for_status()
-        pr_data = response.json()
+        if platform == Platform.GITHUB:
+            data = {
+                'title': final_pr_title,  # No need to escape title for GitHub API
+                'body': pr_body,
+                'head': branch_name,
+                'base': base_branch,
+                'draft': pr_type == 'draft',
+            }
+        # Prepare the PR for the GitLab API
+        elif platform == Platform.GITLAB:
+            data = {
+                'title': final_pr_title,
+                'description': pr_body,
+                'source_branch': branch_name,
+                'target_branch': base_branch,
+                'draft': pr_type == 'draft',
+            }
+
+        pr_data = handler.create_pull_request(data)
+        url = pr_data['html_url'] if platform == Platform.GITHUB else pr_data['web_url']
 
         # Request review if a reviewer was specified
         if reviewer and pr_type != 'branch':
-            review_data = {'reviewers': [reviewer]}
-            review_response = requests.post(
-                f'{base_url}/pulls/{pr_data["number"]}/requested_reviewers',
-                headers=headers,
-                json=review_data,
+            number = (
+                pr_data['number'] if platform == Platform.GITHUB else pr_data['iid']
```

Comment:
NIT: what would be nice would be to avoid too many `if else` statements

The class could have a generic field `pr_number: int`, and `create_pull_request` will open the PR and also populate the class field `pr_number`

That way this line would simply be `number = handler.pr_number` (i.e. the class is responsible for normalizing the data formats into a standard interface)